src/coprocessor/dag/expr/builtin_compare.rs

```diff
@@ -202,6 +202,68 @@ impl ScalarFunc {
     pub fn in_json(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
         do_in(self, |v| v.eval_json(ctx, row), |l, r| Ok(l.cmp(r)))
     }
+
+    pub fn interval_int(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let target = match self.children[0].eval_int(ctx, row) {
+            Err(e) => return Err(e),
+            Ok(None) => return Ok(Some(-1)),
+            Ok(Some(v)) => v,
+        };
+        let tus = mysql::has_unsigned_flag(self.children[0].get_tp().get_flag());
+
+        let mut left = 1;
+        let mut right = self.children.len();
+        while left < right {
+            let mid = left + (right - left) / 2;
+            let m = match self.children[mid].eval_int(ctx, row) {
+                Err(e) => return Err(e),
+                Ok(None) => target,
+                Ok(Some(v)) => v,
+            };
```

Comment:
Could we use `self.children[mid].eval_int(ctx,row)?.unwrap_or(target);` instead?

Reply:
That would be awesome, thank you!